### PR TITLE
Support chunk_min in SmartChunkedIterator

### PR DESCRIPTION
**Pull request by lwiecek**
Fixes #180

### CODE REVIEW

File: django_mysql/models/query.py [hunk 1]
```diff
@@ -246,8 +246,11 @@ def __init__(self, queryset, atomically=True, status_thresholds=None,
         self.pk_range = pk_range
 
         self.rate = WeightedAverageRate(chunk_time)
+        assert 0 < chunk_min <= chunk_max, \
```
Comment (adamchainz) [inline django_mysql/models/query.py]: Don't like line continuations, use the syntax:

```
assert (
    0 < chunk_min <= chunk_max,
    "Minimum...
)
```

File: django_mysql/models/query.py [hunk 2]
```diff
@@ -246,8 +246,11 @@ def __init__(self, queryset, atomically=True, status_thresholds=None,
...
+            "Minimum chunk size should not be greater than maximum chunk size."
+        self.chunk_min = chunk_min
         self.chunk_max = chunk_max
-        self.chunk_size = min(chunk_size, chunk_max)
+        self.chunk_size = self.within_min_and_max(chunk_size)
```
Comment (adamchainz) [inline django_mysql/models/query.py]: the method name is a bit explicit, why not call it `self.constrain_size`?

File: tests/testapp/test_models.py [hunk 3]
```diff
@@ -530,6 +530,22 @@ def test_objects_max_size_bounds_chunk_size(self):
...
+        for chunk in smart:
+            ids = [author.id for author in chunk]
+            assert len(ids) >= 5
+            seen.extend(ids)
+        all_ids = list(Author.objects.order_by('id')
+                                     .values_list('id', flat=True))
+        assert seen == all_ids
+
+    def test_min_size_greater_than_max(self):
+        with self.assertRaises(AssertionError):
```
Comment (adamchainz) [inline tests/testapp/test_models.py]: use

```
with pytest.raises(AssertionError):
```

File: django_mysql/models/query.py (hunk 1 above)
Comment (lwiecek) [inline django_mysql/models/query.py]: assert doesn't like parentheses:
```
>>> assert(False, "abc")
<stdin>:1: SyntaxWarning: assertion is always true, perhaps remove parentheses?
```

Comment (adamchainz) [inline django_mysql/models/query.py]: It does if you leave a space

**Comment by adamchainz** [comment]
Please squash and retitle "Support chunk_min in SmartChunkedIterator".

**Comment by adamchainz** [comment]
Merged in 01437dd1f09ceb5b6f19c68d91a136fc591e682f

**Pull request closed without merging**